tests/core/math/test_aabb.h

```diff
@@ -204,6 +204,30 @@ TEST_CASE("[AABB] Intersection") {
 	CHECK_MESSAGE(
 			!aabb_big.intersects_segment(Vector3(0, 300, 0), Vector3(0, 300, 0)),
 			"intersects_segment() should return the expected result with segment of length 0.");
+
+	CHECK_MESSAGE( // Simple Ray Intersection Test
+			aabb_big.intersects_ray(Vector3(-100, 3, 0), Vector3(1, 0, 0)),
+			"intersects_ray() should return true when ray points directly to AABB from outside.");
+
+	CHECK_MESSAGE( // Ray Parallel to an Edge
+			!aabb_big.intersects_ray(Vector3(10, 10, 0), Vector3(0, 1, 0)),
+			"intersects_ray() should return false for ray parallel and outside of AABB.");
+
+	CHECK_MESSAGE( // Ray Origin Inside AABB
+			aabb_big.intersects_ray(Vector3(1, 1, 1), Vector3(0, 1, 0)),
+			"intersects_ray() should return true for rays originating inside the AABB.");
+
+	CHECK_MESSAGE( // Ray Pointing Away from AABB
+			!aabb_big.intersects_ray(Vector3(-10, 0, 0), Vector3(-1, 0, 0)),
+			"intersects_ray() should return false when ray points away from AABB.");
+
+	CHECK_MESSAGE( // Ray Along a Diagonal of AABB
+			aabb_big.intersects_ray(Vector3(0, 0, 0), Vector3(1, 1, 1).normalized()),
+			"intersects_ray() should return true for rays along the AABB diagonal.");
+
+	CHECK_MESSAGE( // Ray Originating at AABB Edge
+			aabb_big.intersects_ray(aabb_big.position, Vector3(-1, 0, 0)),
+			"intersects_ray() should return true for rays starting on AABB's edge.");
 }
```

Comment:
Should we include unit tests for intersection points and normal?

Reply:
Assuming that marking this as resolved means you leave it to future test writing?